bindings/pyroot/pythonizations/src/CPPInstancePyz.cxx

```diff
@@ -55,35 +55,31 @@ PyObject *PyROOT::CPPInstanceExpand(PyObject * /*self*/, PyObject *args)
       TBufferFile buf(TBuffer::kRead, PyBytes_GET_SIZE(pybuf), PyBytes_AS_STRING(pybuf), kFALSE);
       newObj = buf.ReadObjectAny(0);
    }
-   PyObject *result = BindCppObject(newObj, Cppyy::GetScope(clname));
-   if (result) {
-      // this object is to be owned by the Python interpreter, assuming that the call
-      // originated from there
-      ((CPPInstance *)result)->PythonOwns();
-   }
+   PyObject *result = CPyCppyy::Instance_FromVoidPtr(newObj, clname, /*python_owns=*/ true);
    return result;
 }
 
 /// PyROOT object proxy pickle support
 /// Turn the object proxy instance into a character stream and return for
 /// pickle, together with the callable object that can restore the stream
 /// into the object proxy instance.
-PyObject *op_reduce(CPPInstance *self, PyObject * /*args*/)
+PyObject *op_reduce(PyObject *self, PyObject * /*args*/)
 {
    // keep a borrowed reference around to the callable function for expanding;
    // because it is borrowed, it means that there can be no pickling during the
    // shutdown of the libPyROOT module
    static PyObject *s_expand =
-      PyDict_GetItemString(PyModule_GetDict(PyROOT::gRootModule), const_cast<char *>("_CPPInstance__expand__"));
+      PyDict_GetItemString(PyModule_GetDict(PyROOT::gRootModule), "_CPPInstance__expand__");
 
    // TBuffer and its derived classes can't write themselves, but can be created
    // directly from the buffer, so handle them in a special case
    static Cppyy::TCppType_t s_bfClass = Cppyy::GetScope("TBufferFile");
    TBufferFile *buff = 0;
-   if (s_bfClass == self->ObjectIsA()) {
-      buff = (TBufferFile *)self->GetObject();
+   Cppyy::TCppType_t selfClass = ((CPPInstance*)self)->ObjectIsA();
+   if (selfClass == s_bfClass) {
+      buff = (TBufferFile *)CPyCppyy::Instance_AsVoidPtr(self);
```

Comment:
Maybe let's just use C++ casts ?

Reply:
That's a good idea, and at some point we should modernize the Pythonization library a bit. However, there are so many C casts in this library already (because that's what is also done in CPython), and changing it now would make the style inconsistent. I would prefer to do this in a follow-up PR.